klaytn/test/adapter.test.ts

```diff
@@ -0,0 +1,153 @@
+import { assert } from 'chai'
+import { Requester } from '@chainlink/external-adapter'
+import { assertSuccess, assertError } from '@chainlink/adapter-test-helpers'
+import { AdapterRequest } from '@chainlink/types'
+import { makeExecute } from '../src/adapter'
+
+describe('execute', () => {
+  const jobID = '1'
+  const execute = makeExecute()
+
+  context('successful calls @integration', () => {
+    const requests = [
+      {
+        name: 'id not supplied',
+        testData: {
+          data: {
+            address: '0xf1e73a60687affcb1d6308fcb2ef1da3865ff797',
+            data:
+              '0x000000000000000000000000344d20e71a76b5a44fff62438cd17446a331bf3e70ae84788a565ad436a112ee40d236e6f95106e22bd45cd8b78c8fe8c93c4df0000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000344d20e71a76b5a44fff62438cd17446a331bf3ed204b3fb0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000603d9865000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000',
+            topics: [
+              '0xd8d7ecc4800d25fa53ce0372f13a416d98907a7ef3d8d3bdd79cf4fe75529c65',
+              '0x3836643062336366366262313466306338633237363234316636383361363765',
+            ],
+            value: '{"jobRunID":"1","data":{"result":"1003904"},"result":"1003904"}',
```

Comment:
Is there a reason why the value is structured like our AdapterResponse and not just the value itself?

(e.g. value: "1003904")

Reply:
I used chainlink's external-adapter package to build my test external adapter. Isn't this the common scenario you expect for chainlink?
https://github.com/winnie-byun/demo-klaytn-chainlink/blob/main/api_server_timestamp/index.js

Reply:
The test external adapter looks correct.

For this Klaytn adapter the response structure is made here:
```
    return Requester.success(jobRunID, {
      data: { result },
      result,
      status: 200,
    })
```

Regarding the value param, it should be any information needed to make the request (submit the Klaytn transaction). Since it seems like all we use is the "1003904" in:
```

  const params = caver.abi
    .encodeParameters(
      ['bytes32', 'uint256', 'address', 'bytes4', 'uint256', 'bytes32'],
      [
        decoded.requestId,
        decoded.payment,
        decoded.callbackAddr,
        decoded.callbackFunctionId,
        decoded.cancelExpiration,
        caver.utils.leftPad(value, 64), //////////// HERE
      ],
    )
    .substring(2)
```

Reply:
No there is no such structure and "1003904" is all we use.

I meant when the values are from [another external-adapter](https://github.com/winnie-byun/demo-klaytn-chainlink/blob/main/api_server_timestamp/index.js), what I see in klaytn external-adapter is : 
```
   value: '{"jobRunID":"1","data":{"result":"10030019"},"result":"10030019"}'
```
Is there some kind of API to get only the value?

Reply:
Oh, you are planning to chain together two external adapters right?

We call those `composite adapters`. They can be seen in the [composite folder](https://github.com/smartcontractkit/external-adapters-js/tree/develop/composite). Currently they are undergoing a large refactor so that the composite adapter sends requests to the underlying external adapters.

It would look like:
- a new composite adapter that will...
- gets the time from `api_server_timestamp` 
- parse the value from the previous response and use it as `value` to request the `klaytn` adapter

[This refactored composite adapter](https://github.com/smartcontractkit/external-adapters-js/pull/349) might serve as an example until we finalize the refactor and show an example composite adapter.